controllers/openstackmachine_controller.go

```diff
@@ -334,9 +359,26 @@ func (r *OpenStackMachineReconciler) reconcileNormal(ctx context.Context, scope
 		return ctrl.Result{}, err
 	}
 
-	instanceStatus, err := r.getOrCreate(scope.Logger(), cluster, openStackCluster, machine, openStackMachine, computeService, userData)
+	var machinePortsStatus []infrav1.PortStatus
+	if openStackMachine.Status.PortsStatus != nil {
```

Comment:
Question for reviewers: are we fine if the user removed ports themselves but it'll be in Status? (note that's the current behaviour for now, as the ports won't get re-created if a user removed them).
If the answer is "no", then we'll have to do calls to Neutron to check that ports defined in PortsStatus actually exist.

Reply:
The user shouldn't be deleting ports. The reasonable way to recover from that would be deleting the Machine.

I'm personally ok with not checking that ports still exist.